src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/CheckCoscMarkPaidCallbackHandler.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.civil.handler.callback.camunda.payment;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.Callback;
+import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.service.Time;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+import static java.util.Optional.ofNullable;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.SUBMITTED;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.CHECK_PAID_IN_FULL_SCHED_DEADLINE;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CheckCoscMarkPaidCallbackHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = singletonList(CHECK_PAID_IN_FULL_SCHED_DEADLINE);
+    private final ObjectMapper objectMapper;
+    private final Time time;
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT), this::checkIsMarkedPaid,
+            callbackKey(SUBMITTED), this::emptySubmittedCallbackResponse
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse checkIsMarkedPaid(CallbackParams callbackParams) {
+
+        CaseData caseData = callbackParams.getCaseData();
+        CaseData.CaseDataBuilder dataBuilder = caseData.toBuilder();
+        if (caseData.getActiveJudgment() == null || ofNullable(caseData.getActiveJudgment().getFullyPaymentMadeDate()).isEmpty()) {
+            caseData.setCoscSchedulerDeadline(time.now().plusDays(30));
+        }
+
```

Comment:
coscApplicationStatus needs to be added here and updated, - should be set it to active.